modules/apps/layout/layout-utility-page/layout-utility-page-sign-in/src/main/java/com/liferay/layout/utility/page/sign/in/internal/struts/SignInStrutsAction.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.layout.utility.page.sign.in.internal.struts;
+
+import com.liferay.petra.io.unsync.UnsyncStringWriter;
+import com.liferay.portal.kernel.model.LayoutSet;
+import com.liferay.portal.kernel.service.LayoutSetLocalService;
+import com.liferay.portal.kernel.servlet.PipingServletResponse;
+import com.liferay.portal.kernel.servlet.ServletResponseUtil;
+import com.liferay.portal.kernel.struts.StrutsAction;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.theme.ThemeUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Olivér Kecskeméty
+ */
+@Component(property = "path=/portal/sign_in", service = StrutsAction.class)
+public class SignInStrutsAction implements StrutsAction {
```

Comment:
Did you consider adding the logic from this Struts Action into the existing `LoginAction`?
Then there would not be a new DXP path to support. Just curious about the reasoning.

Reply:
Yes, and István also raised this question. I decided to follow the pattern as the 500 status and the Terms of Use utility pages also defined struts actions in their modules.